Launcher/Launcher/src/Helpers/InstallationHelper.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Reflection;
+using WSL;
+
+namespace Launcher.Helpers;
+
+public static class InstallationHelper {
+    /// <summary>
+    ///     Registers the distribution and runs first-time setup
+    /// </summary>
+    /// <returns>0 on success, an error code otherwise</returns>
+    public static int Install() {
+        Console.WriteLine($"Registering {DistributionInfo.DisplayName}...");
+
+        if (WslApiLoader.WslIsDistributionRegistered(DistributionInfo.Name)) {
+            Console.Error.WriteLine($"{DistributionInfo.DisplayName} is already installed!");
+            return 0;
+        }
+
+        // Determine tarball location
+        var tarPath = FindTarball();
+        if (tarPath == null) {
+            Console.Error.WriteLine("Could not find distro tarball");
+            return 1;
+        }
+
+        try {
+            WslApiLoader.WslRegisterDistribution(
+                DistributionInfo.Name,
+                tarPath
+            );
+        } catch (WslApiException e) {
+            Console.Error.WriteLine("There was an error registering the distribution");
+            return e.HResult;
+        }
+
+        Console.WriteLine("Performing first-time setup...");
+
+        if (!StartupHelper.BootDistro()) {
+            Console.Error.WriteLine("An error occured during first-time setup");
+            return 1;
+        }
+
+        Console.WriteLine("Installation finished successfully");
+        return 0;
+    }
+
+    /// <summary>
+    ///     Unregister the distribution
+    /// </summary>
+    /// <returns>0 on success and an error code otherwise</returns>
+    public static int Uninstall() {
+        Console.WriteLine($"Uninstalling {DistributionInfo.DisplayName}...");
+        if (!WslApiLoader.WslIsDistributionRegistered(DistributionInfo.Name)) {
+            Console.Error.WriteLine($"{DistributionInfo.DisplayName} is not installed!");
+            return 1;
+        }
+
+        try {
+            WslApiLoader.WslUnregisterDistribution(DistributionInfo.Name);
+        } catch (WslApiException e) {
+            Console.Error.WriteLine("An error occured when unregistering the distribution!");
+            return e.HResult;
+        }
+
+        Console.WriteLine("Uninstall completed");
+        return 0;
+    }
+
+    /// <summary>
+    ///     Find the path of the installer tarball
+    /// </summary>
+    /// <returns>the full path to the tarball or null</returns>
+    public static string? FindTarball() {
+        const string tarFileName = "nixos-wsl-installer.tar.gz";
+
+        var tarPath = "";
```

Comment:
As far as I know, this shouldn't make a difference

Reply:
Yeah, I should have flagged this as a nit as well. I usually make this recommendation because it's more clear to any readers that you intended this to be an empty string.

Reply:
I've removed that line entirely, so this isn't applicable anymore